Rilmodem GPRS tests

alfonsosanchezbeato opened the pull request:
Rilmodem GPRS tests. The PR includes a rewrite of our testing framework.

tonyespy commented on unit/test-rilmodem-cv.c:
```diff
@@ -0,0 +1,463 @@
+#include "ril_constants.h"
+#include "rilmodem-test-engine.h"
+
+static const struct ofono_call_volume_driver *cvriver;
```
Why is this names "cvriver" instead of "cvdriver"?

tonyespy commented on unit/test-rilmodem-devinfo.c:
```diff
@@ -0,0 +1,355 @@
+	struct ofono_devinfo *devinfo = g_malloc0(sizeof(*devinfo));
+	int retval;
+
+	devinfo->rdid = rdid;
```
From what I can see, this struct member never gets used...

tonyespy commented on unit/test-rilmodem-gprs.c:
```diff
@@ -0,0 +1,744 @@
+#include "ril_constants.h"
+#include "rilmodem-test-engine.h"
+
+static const struct ofono_gprs_driver *gprs_drv;
```
Super nit-pick, but we've been using "atom-name" | "driver" ( 'cbdriver', 'cfdriver', ... ).

alfonsosanchezbeato commented:
@tonyespy thanks for the review!

Re: pull request, I understand, but having different PRs **if there are dependencies** does not really provide a big advantage, because in the end the newest includes the code from the older ones. In fact I had two PRs for these changes and as you point out I refactored call-volume in the second one, while if having just one PR I would have just rebased and made your review easier. Bazaar was nicer on that regard I think. Sorry for the mess anyway.

I have addressed the comments in https://github.com/rilmodem/ofono/pull/254 (good catches). Only thing I have not done is changing the name gprs_drv, because that file was already accepted upstream and it is not worth keeping a patch just for this. And hey, you said it was super nit-pick ;-)
